webonary-cloud-api/lambda/methodAuthorize.ts

```diff
@@ -33,47 +35,42 @@ export async function handler(
   context.callbackWaitsForEmptyEventLoop = false;
 
-  const dictionaryId = event.pathParameters?.dictionaryId;
   const authHeaders = event.headers?.Authorization;
+  const dictionaryId = event.pathParameters?.dictionaryId;
 
   if (dictionaryId && authHeaders) {
-    try {
-      const encodedCredentials = authHeaders.split(' ')[1];
-      const plainCredentials = Buffer.from(encodedCredentials, 'base64')
-        .toString()
-        .split(':');
-      const username = plainCredentials[0];
-      const password = plainCredentials[1];
+    const credentials = getBasicAuthCredentials(authHeaders);
 
-      // Same user should have the same access to post dictionary entry data as well as files.
-      // User access is per dictionary per user.
-      const principalId = `${dictionaryId}::${username}`;
+    console.log(`Processing policy for ${credentials.username} to resource ${event.methodArn}`);
+    // Same user should have the same access to post/delete dictionary entry data as well as files.
+    const principalId = credentials.username;
+    const resourceRegex = /(POST\/post|DELETE\/delete)\/(dictionary|entry|file)\/(.+)$/i;
 
-      // To allow for correct caching behavior, we use wildcards for method (POST or DELETE) and path
-      const resource = event.methodArn.replace(
-        /(POST\/post|DELETE\/delete)\/(dictionary|entry|file)\//i,
-        '*/*/',
-      );
-
-      // Call Webonary.org for user authentication
-      axios.defaults.headers.post['Content-Type'] = 'application/json';
-      const authPath = `${process.env.WEBONARY_URL}/${dictionaryId}${process.env.WEBONARY_AUTH_PATH}`;
+    // Call Webonary.org for user authentication
+    axios.defaults.headers.post['Content-Type'] = 'application/json';
+    const authPath = `${process.env.WEBONARY_URL}/${dictionaryId}${process.env.WEBONARY_AUTH_PATH}`;
 
+    try {
       const response = await axios.post(authPath, '{}', {
-        auth: { username, password },
+        auth: credentials,
       });
 
-      console.log(`Processing policy for ${username} to resource ${event.methodArn}`);
-      if (response.status === 200) {
-        if (response.data) {
-          console.log(`Denied ${principalId} to resource ${resource} for ${response.data}`);
-          return callback(null, generatePolicy(principalId, 'Deny', resource)); // 403
-        }
-        console.log(`Created policy for ${principalId} to access resource ${resource}`);
-        return callback(null, generatePolicy(principalId, 'Allow', resource));
+      if (response.status === 200 && response.data) {
+        const resources = response.data.split(',').map((id: string) => {
+          // To allow for correct caching behavior, we use wildcards for method (POST or DELETE) and path
+          // arn:aws:execute-api:region:zz:zzz/prod/POST/post/dictionary/myDictionary will be replaced with
+          // arn:aws:execute-api:region:zz:zzz/prod/*/*/dictionary/myDictionary will be replaced with
```

Comment:
I think you left the 'will be replaced with' after a copy paste. :)

Reply:
ah, fixed. thank you.